src/native/minipal/cpufeatures.c

```diff
@@ -306,6 +306,42 @@ int minipal_getcpufeatures(void)
                 result |= XArchIntrinsicConstants_Serialize;                                               // SERIALIZE
             }
         }
+
+        if (maxCpuId >= 0x07)
+        {
+            // Detect Avx10 here.
+            // Given paradigm:
+            // 1. CPUID.(EAX=07H, ECX=01H):EDX[bit 19] = 1 -> ISA supported.
+            // 2. CPUID.(EAX=24H, ECX=00H):EBX[bits 7:0] >= 1 -> ISA version larger than 1.
+            // with 1,2, set XArchIntrinsicConstants_Avx10_1
+
+            // 3. CPUID.(EAX=24H, ECX=00H):EBX[bits 18:16] -> vector length support.
+            __cpuidex(cpuidInfo, 0x00000007, 0x00000001);
```

Comment:
I think we can put this further up, where we already query `__cpuidex(cpuidInfo, 0x00000007, 0x00000001)` as part of `AvxVnni`

That path is after the `Avx2` check and so its only happening after we've already confirmed other required ISAs are supported.